### PR TITLE
OPSEXP-1584 Repository Volume Without Default Alfresco Volume Claim

### PR DESCRIPTION
**Pull request by upandey123**
Kindly refer the files changes in this link,

https://github.com/Alfresco/acs-deployment/pull/820/files

### CODE REVIEW

File: helm/alfresco-content-services/templates/pvc.yaml [hunk 1]
```diff
@@ -22,3 +22,7 @@ spec:
 {{- if and (not .Values.filestore.persistence.existingClaim) (eq .Values.repository.edition "Enterprise") .Values.filestore.persistence.enabled }}
```
Comment (alxgomz) [inline helm/alfresco-content-services/templates/pvc.yaml:22]: There's something wrong with your PR. The diff should not look like that given current status of the [`pvc.yaml` in master](https://github.com/Alfresco/acs-deployment/blob/99635eacb1601236f174d42c0710e44a45662af6/helm/alfresco-content-services/templates/pvc.yaml)... Please fix as doing a review is too difficult currently

File: helm/alfresco-content-services/templates/pvc.yaml [hunk 2]
```diff
@@ -22,3 +22,7 @@ spec:
...
 {{ include "component_pvc" .Values.filestore }}
 {{- end }}
+
+{{- if and (not .Values.repository.persistence.existingClaim) (eq .Values.repository.edition "Enterprise") .Values.repository.persistence.enabled }}
+{{ include "component_pvc" .Values.repository }}
+{{- end }}
```
Comment (alxgomz) [inline helm/alfresco-content-services/templates/pvc.yaml:28]: preserve order of resources as unit tests may rely on it

File: helm/alfresco-content-services/values.yaml [hunk 3]
```diff
@@ -525,25 +525,39 @@ share:
 # -- Defines the mounting points for the persistence required by the apps in the
 # cluster the alf_data folder from alfresco-content-repository app is mapped to
 # alfresco-content-services/repository-data
+#persistence:
```
Comment (alxgomz) [inline helm/alfresco-content-services/values.yaml:528]: prefer cleaning over commenting

File: helm/alfresco-content-services/values.yaml [hunk 4]
```diff
@@ -525,25 +525,39 @@ share:
...
+#      querysetsMountPath: "/usr/local/tomcat/shared/classes/alfresco/extension/querysets/"
+#      transform:
+#        renditionsMountPath: "/usr/local/tomcat/shared/classes/alfresco/extension/transform/renditions/"
+#        pipelinesMountPath: "/usr/local/tomcat/shared/classes/alfresco/extension/transform/pipelines/"
+#        mimetypesMountPath: "/usr/local/tomcat/shared/classes/alfresco/extension/mimetypes/"
+#    data:
+#      mountPath: "/usr/local/tomcat/alf_data"
+#      subPath: "alfresco-content-services/repository-data"
+
 persistence:
```
Comment (alxgomz) [inline helm/alfresco-content-services/values.yaml:551]: there should not be any root `persistence` data structure after this PR.

**Review by alxgomz — CHANGES_REQUESTED** [review]
the helper has been moved to the alfresco-common chart you must not add it again

File: helm/alfresco-content-services/templates/deployment-repository.yaml [hunk 5]
```diff
@@ -203,7 +203,14 @@ spec:
             - name: email-keystore-volume
               mountPath: /var/run/secrets/java.io/keystores
         {{- end }}
+          volumeMounts:
```
[the remark this answers is not in the document]
Comment (upandey123) [reply helm/alfresco-content-services/templates/deployment-repository.yaml]: fixed

File: helm/alfresco-content-services/templates/pvc.yaml (hunk 1 above)
Comment (upandey123) [reply helm/alfresco-content-services/templates/pvc.yaml]: fixed

File: helm/alfresco-content-services/templates/pvc.yaml (hunk 2 above)
Comment (upandey123) [reply helm/alfresco-content-services/templates/pvc.yaml:28]: fixed

File: helm/alfresco-content-services/values.yaml (hunk 3 above)
Comment (upandey123) [reply helm/alfresco-content-services/values.yaml]: fixed

File: helm/alfresco-content-services/values.yaml (hunk 4 above)
Comment (upandey123) [reply helm/alfresco-content-services/values.yaml:529]: fixed

**Comment by upandey123** [comment]
@alxgomz @gionn Do we need to have some values in values.yaml inside alfresco-common ? I just created the empty file because it was showing up error. Now alfresco-common check is failing because it could not find values there it seems.
Also, could you please have a look on my deployment-repository.yaml file from line 112 to 116. All the other checks are failing because of no value to key.

https://github.com/Alfresco/acs-deployment/actions/runs/3618651161/jobs/6098761405

**Comment by alxgomz** [comment]
No, the `alfresco-common` chart is a library chart it doesn't need a value file and the presence of this value file is making the process try to install it which cannot be done. Remove that file (is will simply trigger a warning you can ignore)

File: helm/alfresco-content-services/templates/deployment-repository.yaml [hunk 6]
```diff
@@ -204,6 +209,8 @@ spec:
               mountPath: /var/run/secrets/java.io/keystores
         {{- end }}
       volumes:
+      {{- include "data_volume" .Values.repository | nindent 8 }}
+       {{- end }}
```
Comment (gionn) [inline helm/alfresco-content-services/templates/deployment-repository.yaml:213]: needs to be wrapped with `{{- if .Values.repository.persistence.enabled }}`

check the end that is not indented correctly

File: helm/alfresco-content-services/templates/deployment-repository.yaml [hunk 7]
```diff
@@ -253,4 +257,3 @@ spec:
 {{ toYaml .Values.repository.extraVolumes | indent 6 }}
       {{- end }}
       {{- end }}
-      {{- end }}
```
Comment (gionn) [inline helm/alfresco-content-services/templates/deployment-repository.yaml:256]: to revert?

File: helm/alfresco-content-services/values.yaml (hunk 4 above)
Comment (gionn) [reply helm/alfresco-content-services/values.yaml:542]: that means we expect a removal here

**Review by gionn — CHANGES_REQUESTED** [review]

File: helm/alfresco-content-services/templates/deployment-repository.yaml (hunk 6 above)
[the remark this answers is not in the document]
Comment (upandey123) [reply helm/alfresco-content-services/templates/deployment-repository.yaml:213]: Applied

    volumes:
209       {{- if .Values.repository.persistence.enabled }}
210       {{- include "data_volume" .Values.repository | nindent 8 }}
211       {{- end }}

File: helm/alfresco-content-services/templates/deployment-repository.yaml [hunk 8]
```diff
@@ -82,6 +82,11 @@ spec:
 {{ toYaml .Values.repository.resources | indent 12 }}
           {{- if .Values.persistence.repository.enabled }}
           volumeMounts:
+          {{- if .Values.repository.persistence.enabled }}
+          - name: data
```
[the remark this answers is not in the document]
Comment (upandey123) [reply helm/alfresco-content-services/templates/deployment-repository.yaml:86]: done

File: helm/alfresco-content-services/values.yaml [hunk 9]
```diff
@@ -92,6 +92,19 @@ repository:
     initialDelaySeconds: 130
     periodSeconds: 20
     timeoutSeconds: 10
+  persistence:
+    # -- Persist repository data
+    enabled: true
+    # -- Specify a storageClass for dynamic provisioning
+    accessModes:
+      - ReadWriteOnce
```
[the remark this answers is not in the document]
Comment (upandey123) [reply helm/alfresco-content-services/values.yaml:100]: done

File: helm/alfresco-content-services/templates/deployment-repository.yaml (hunk 7 above)
Comment (upandey123) [reply helm/alfresco-content-services/templates/deployment-repository.yaml:256]: done

File: helm/alfresco-content-services/templates/deployment-repository.yaml (hunk 6 above)
Comment (upandey123) [reply helm/alfresco-content-services/templates/deployment-repository.yaml:213]: done like this,

209 {{- if .Values.repository.persistence.enabled }}
210 {{- include "data_volume" .Values.repository | nindent 8 }}
211 {{- end }}

File: helm/alfresco-content-services/templates/pvc.yaml [hunk 10]
```diff
@@ -19,6 +19,10 @@ spec:
       storage: {{ .Values.persistence.baseSize | quote }}
   {{- end -}}
 
+{{- if and (not .Values.repository.persistence.existingClaim) .Values.repository.persistence.enabled }}
```
[the remark this answers is not in the document]
Comment (upandey123) [reply helm/alfresco-content-services/templates/pvc.yaml:22]: deleted the previous pvc

del1-lmc-n02498:alfresco-content-services utkarshpandey$ cat templates/pvc.yaml 
{{- if .Values.persistence.enabled -}}
kind: PersistentVolumeClaim
apiVersion: v1
metadata:
  name: alfresco-volume-claim
spec:
 {{- if and (not .Values.repository.persistence.existingClaim) .Values.repository.persistence.enabled }}
 {{ include "component_pvc" .Values.repository }}
 {{- end }}

 {{- if and (not .Values.filestore.persistence.existingClaim) (eq .Values.repository.edition "Enterprise") .Values.filestore.persistence.enabled }}
 {{ include "component_pvc" .Values.filestore }}
 {{- end }}
del1-lmc-n02498:alfresco-content-services utkarshpandey$

File: helm/alfresco-content-services/values.yaml [hunk 11]
```diff
@@ -92,6 +92,19 @@ repository:
...
+    # -- Bind PVC based on storageClass (e.g. dynamic provisionning)
+    storageClass:
+    # -- Use pre-provisioned pv through its claim (e.g. static provisionning)
+    existingClaim: null
+    data:
+      mountPath: "/tmp/Alfresco"
+      subPath: "alfresco-content-services/repository-data"
```
[the remark this answers is not in the document]
Comment (upandey123) [reply helm/alfresco-content-services/values.yaml:107]: done

**Comment by upandey123** [comment]
@alxgomz The init containers of postgres are not starting up. The community and enterprise charts also failing because of that. Could you please have a look and let me know ?

**Review by slohe1 — APPROVED** [review]
LGTM

**Review by alxgomz — APPROVED** [review]

**Pull request merged**